pkg/dao/types/resource_run.go

```diff
@@ -134,3 +124,152 @@ type Change struct {
 
 	Type string `json:"type"`
 }
+
+// Patch patch the change and return the patched change.
+// The patching includes setting the change type, the masking of the sensitive values and the unknown values.
+func (c *Change) Patch() *Change {
+	switch {
+	case c.Actions.Create():
+		c.Type = ResourceComponentChangeTypeCreate
+	case c.Actions.Update(),
+		c.Actions.Replace():
+		c.Type = ResourceComponentChangeTypeUpdate
+	case c.Actions.Delete():
+		c.Type = ResourceComponentChangeTypeDelete
+	case c.Actions.NoOp(), c.Actions.Read():
+		c.Type = ResourceComponentChangeTypeNoChange
+	}
+
+	c.Change.Before = patchLeaf(c.Change.Before, c.Change.BeforeSensitive, "<sensitive>")
+	c.Change.After = patchLeaf(c.Change.After, c.Change.AfterSensitive, "<sensitive change>")
+	c.Change.After = patchLeaf(c.Change.After, c.Change.AfterUnknown, "<after known>")
```

Comment:
Why use <after known> instead of <after unknown>?

Reply:
Let's clear it with `known after apply`